CONTRIBUTING.md

```diff
@@ -60,9 +59,10 @@ our [Style Guide](docs/StyleGuide.md).
       then `git rebase origin/master`) to make sure you're
       submitting your changes on top of the newest version of our code.
 
-    * When finalizing your PR, you may squash commits in order to ensure that
-      each commit represents a clean, logical change and includes a
-      descriptive commit message.  
+    * When finalizing your PR, aim for the fewest number of commits that
+      make sense, i.e., squash any "fix up" commits into the commit they
```

Comment:
While this makes sense for the final version, it does make reviewing PRs more difficult, as it can be difficult to find what's changed since the last version. Can we have the convention be that we squash before the PR is merged but only after it is accepted?

Reply:
Understandable. I think the two times you want this to be true are (1) when it is first submitted for review and (2) right before merge (as you said).

I can change to `When finalizing your PR (i.e., it has been approved for merging), aim for the fewest...`